scipy/datasets/_fetchers.py

```diff
@@ -27,8 +27,10 @@ def fetch_data(dataset_name, data_fetcher=data_fetcher):
         raise ImportError("Missing optional dependency 'pooch' required "
                           "for scipy.datasets module. Please use pip or "
                           "conda to install 'pooch'.")
+    # https://github.com/scipy/scipy/issues/21879
+    downloader = pooch.HTTPDownloader(headers={"User-Agent": "agent"})
```

Comment:
Another thing to consider is, if we should name the agent explicitly indicating SciPy or keep the dummy name. I'm not sure about it though (cc @lucascolley @rgommers any suggestions?):

Maybe something like?
```headers = {"User-Agent": f"SciPy {sys.modules['scipy'].__version__}"}```

Reply:
This suggestion SGTM. Disclaimer: I'm not good at web stuff, so if if there are any hidden implications, I won't be the one catching them here.

Reply:
Thanks, @tkoyama010 let's go with the above suggested header naming with the SciPy version for the `_fetchers.py`. But for `_dowload_all.py`, i suggest we just keep the header user-agent name as "SciPy", since that function is also callable via the standalone script prior to building SciPy which means the version info won't be available if the user tries to run `python _download_all.py <download_dir>`.

Reply:
```suggestion
    downloader = pooch.HTTPDownloader(headers={"User-Agent": f"SciPy {sys.modules['scipy'].__version__}"})
```